### ISSUE TITLE
Getters and setters returning wild values

### ISSUE THREAD

**Issue by inhalt120g** [godotengine/godot#100928]
### Tested versions

v4.3.stable.official [77dcf97d8]

### System information

Newest MacOS on Mac Mini M1

### Issue description

Values of setters and getters produce wild values under certain circumstances.

### Steps to reproduce

Download the MRP.
Confirm the only code in the project is in the base model of the item:

```
extends Node2D

@export var item_worth : float = 10:
	set(new_v): # new value
		item_worth = new_v
		a_prices.gold_total += item_worth
		print("New total: ", a_prices.gold_total)
```

Open the "room" scene, and click on the items (white rectangles) and confirm in the inspector their values are 10 and 100.

(At this point… Can you guess what is the total gold value you'll get after running the example?
I'd expect no output whatsoever, because I set the values in the editor and this should not trigger the setter at all. Next logical expectation would be that the total value would be 110 since that's what the items have in their "item_worth", 100+10 = 110, but this is also not going to happen.)

Run the example and check the console.
The total value you'll get from the two items is 140!

To me this looks like a bug, but I expect the explanation behind this behavior will be that the values entered in the editor are applied when the game is run (including the modification to the base model). This is, however, not expected at all in this context unless you know it in advance. Im sure there are other reasons why it acts like this, but I was not able to find it mentioned anywhere in the docs. If this is a feature, it should at least be clarified somewhere in the docs.

### Minimal reproduction project (MRP)

[engage-funny-setter-geter.zip](https://github.com/user-attachments/files/18275416/engage-funny-setter-geter.zip)

**Comment by CodeZombie** [comment]
> I'd expect no output whatsoever, because I set the values in the editor and this should not trigger the setter at all.

A scene file (`.tcsn`) is basically just a data structure file, like a `.json` or `.yaml`. When you modify values in the inspector, the Godot editor saves these changes in this data structure file. In the case of `room.tcsn`, it stores:
```
[ext_resource type="PackedScene" uid="uid://prnvwtmf8kt5" path="res://item 01.tscn" id="1_vhosw"]

[node name="item 01 slot" parent="." instance=ExtResource("1_vhosw")]
position = Vector2(271, 172)
item_worth = 10.0

[node name="item 01 slot2" parent="." instance=ExtResource("1_vhosw")]
position = Vector2(405, 176)
item_worth = 100.0
```
The contents of this file have zero impact on your code until you instantiate the scene at runtime, at which point the Godot engine reads the `.tcsn` file and assigns the values it finds to their respective locations in your code.

Because you've defined a `set` method for `item_worth`, Godot is forced to use this method to assign values, and thus, your setter gets run at runtime. However, **if you don't change the value of a variable in the inspector, it won't have anything to assign, and thus, won't run your setter when instantiating the scene**

### Why gold_total is weird

> I expect the explanation behind this behavior will be that the values entered in the editor are applied when the game is run (including the modification to the base model).

Yep.

When Godot instantiates room.tcsn, it also has to instantiate two `item 01.tcsn`s, which are really just `base item.tcsn`.
Inside of `base item.tcsn` we can see that you've edited the `item_worth` variable to `15`. 
Because of this, any time `base item.tcsn` (or anything that inherits from it) gets instantiated, Godot needs to send the values it finds in this data file (`item_worth = 15`) into the code memory. And thus, your setter runs.
But `room.tcsn` isn't done yet. We still have `item_worth` values in here (`10` and `100`). So the engine sends those values to the code memory via your setter as well, and thus, the set method is getting run again, twice for each item:

tl;dr, instantiating `room.tcsn` triggers the following behavior:
1. `item 01 slot` (instance of `base item.tcsn`) gets instantiated, where `item worth` is overridden to `15`, so `set(15)` gets run.
2. `item 01 slot`'s `item_worth` is set to `10` so `set(10)` runs.
3. `item 01 slot2`  (instance of `base item.tcsn`) gets instantiated, where `item worth` is overridden to `15`, so `set(15)` gets run.
4. `item 01 slot2`'s `item_worth` is set to `100`, so `set(100)` runs.

`15 + 10 + 15 + 100 = 140`

If you don't override `item_worth` to 15 in `base item.tcsn`, those two `set(15)`s won't need to get run, and thus, your `gold_total` will be `110`, as you expect.

This isn't so much a bug as it is a limitation of the fundamental architecture of Godot. I agree though that the documentation should be more transparent about this design.

**Comment by inhalt120g** [comment]
Thank you for the detailed breakdown, I guessed this must be going on by observing the values, but it's not fun discovering this on my own in the middle of the project. Hopefully the post above will be of help to other people who might bump into this problem.
